### ISSUE TITLE
Accessibility - audible feedback on task completion

### ISSUE THREAD

**Issue by psychemedia** [tqdm/tqdm#988]
- [x] I have marked all applicable categories:
    + [x] documentation request (i.e. "X is missing from the documentation." If instead I want to ask "how to use X?" I understand [StackOverflow#tqdm] is more appropriate)
    + [x] new feature request

I'm working in an educational Jupyter notebook context and am keen to make notebooks as accessible as possible.

To this end, it would be useful to be able to provide feedback to partially sighted or blind users as well as sighted users when using TQDM.

For example:

- generate a audible alert when a task bar completes;
- generate an increasing or decreasing tone as an audible correlate to the animated task bar (perhaps optionally prefixed by a tone representing the frequency of the tone you'd expect to hear when the task completes to prime the listener).

Audible feedback might also be useful for sighted users who maybe go and work in another part of the notebook while executing a long running TQDM reported task in another cell in an out-of-sight area of the notebook.

I'm not sure how easy such feedback would be in the general package, but in the Jupyter notebook variant there is ready access to audio players, for example via the `from IPython.display import Audio` player or by Javascript embedding using `from IPython.display import Audio`. 

In the first instance, I guess that adding a callback to TQDM could provide a task completion tone, or perhaps a spoken alert that announces task completion:

```python
from IPython.display import Javascript

def say(txt):
        """Speak an utterance."""
        js = f'''
        var utterance = new SpeechSynthesisUtterance("{txt}");
        '''
        js = js + 'speechSynthesis.speak(utterance);'
        display(Javascript(js))

say('All done')
```

Are there any examples of this sort of support using TQDM anywhere?

**Comment by casperdcl** [comment]
I could write a quick example but first - is there a nice bell sound built in instead (e.g. like a kitchen timer chiming)? That would be more universal/langage - agnostic.

On a realted note, I can also implement a bell on the terminal.

**Comment by psychemedia** [comment]
@casperdcl There are some examples of simple tones generated in Javascript [here](https://marcgg.com/blog/2016/11/01/javascript-audio/).

Cribbing the code from some one of the examples:

```python
js = '''
function example4(frequency, type) {
    o = context.createOscillator()
    g = context.createGain()
    o.type = type
    o.connect(g)
    o.frequency.value = frequency
    g.connect(context.destination)
    o.start(0)
    g.gain.exponentialRampToValueAtTime(0.00001, context.currentTime + 1)
}
'''

display(Javascript(tone+"example4(440, 'square')"))
```

**Comment by psychemedia** [comment]
@casperdcl Re: the spoken feedback, I was thinking the user could perhaps provide an explicit spoken message of their own, eg if they wanted to distinguish several different tasks completing?

Being able to specify different tones via a simple optional parameter might also be useful?

**Comment by casperdcl** [comment]
well a simple way of doing this is:

```python
from tqdm.notebook import tqdm as tqdm_notebook_orig


class tqdm_notebook(tqdm_notebook_orig):
    def close(self, *args, **kwargs):
        if self.disable:
            return
        super(tqdm_notebook, self).close(*args, **kwargs)

        from IPython.display import Javascript, display
        display(Javascript('''
        var context = new AudioContext()
        function tone(duration=1.5, frequency=400, type='sin'){
          var o = context.createOscillator(); var g = context.createGain()
          o.frequency.value = frequency; o.type = type
          o.connect(g); g.connect(context.destination)
          o.start(0)
          g.gain.exponentialRampToValueAtTime(0.00001, context.currentTime + duration)
        }
        tone(1.5, 600)'''))
```

any you can test with:

```python
for i in tqdm_notebook(range(10)):
    pass
```

**Comment by psychemedia** [comment]
Great, thanks... will riff on that idea...

**Comment by casperdcl** [comment]
yes you could override `__init__()` to add optional `duration=1.5, frequency=400, type='sin'`. Not sure if this is something to put into [`tqdm.contrib`](https://github.com/tqdm/tqdm/tree/master/tqdm/contrib) or the [`examples/`](https://github.com/tqdm/tqdm/tree/master/examples) folder.

**Comment by psychemedia** [comment]
Overriding would certainly give the user a bit more flexibility. Would it be worth treating as an example first to see if generates any additional comments/user examples before turning it into a `contrib` solution?
